### PR TITLE
UX improvement of project parameter administration

### LINKED ISSUE #30: Enhance the usability of the edition of Audit parameters
Here are a few suggestions to enhance the edition of audit parameters:

- [ ] When I add an audit parameter, if I click on the configuration select list, I should not exit the edition mode;
- [ ] When I add an audit parameter, if I hit <kbd>Enter</kbd> my input is submitted
- [ ] When I add an audit parameter, the new line in the table is aligned with existing ones (see screenshot).

image

@alexfauquette would you be willing to work on this? @LouisPinsard can help you navigate this corner of the codebase :)

### PR DESCRIPTION
**Pull request by alexfauquette**
Closes #30 

For now it's only on the "environment project". Once we agree on the technic, I will extend it to the "page of the project".

image

@phacks I noticed that in the data row, data width sum to 100% more 5% for the delete button. In e responsive point of view, I think it's better to fix button width and let input grow as they want.

### CODE REVIEW

File: frontend/src/icons/Done/Done.tsx [hunk 1]
```diff
@@ -0,0 +1,29 @@
+import React from 'react';
+
+interface Props {
+  color: string;
+  width?: string;
+  height?: string;
+  strokeWidth?: string;
+}
+
+export default class Done extends React.Component<Props> {
```
Comment (phacks) [inline frontend/src/icons/Done/Done.tsx]: I think `Checkmark` would be a better name

File: frontend/src/pages/EnvironmentSettings/Components/AuditParameterTable/AddAuditParameterRow/AddAuditParameterRow.tsx [hunk 2]
```diff
@@ -1,9 +1,13 @@
 import Select from 'components/Select/Select';
-import { Add } from 'icons';
+import Add from 'icons/Add';
```
Comment (phacks) [inline frontend/src/pages/EnvironmentSettings/Components/AuditParameterTable/AddAuditParameterRow/AddAuditParameterRow.tsx]: Can these be imported as `AddIcon`, `CloseIcon` and `DoneIcon`? I would make it clear that they are merely icons, not full-fledged components.

You can do `import Add as AddIcon from 'icons/Add'`

File: frontend/src/pages/EnvironmentSettings/Components/AuditParameterTable/AuditParameterTable.style.tsx [hunk 3]
```diff
@@ -137,3 +138,10 @@ export const AuditParameterDeleteButton = styled.button`
     transition-duration: 0.2s;
   }
 `;
+
+export const AddAuditParameterButtonsContainer = styled.div`
+  display: ${(props: AddInputProps) => props.isAdding ? "flex" : "none"};
+  width: 60px;
+  flex-direction: row;
+  justify-content: center;
+`;
```
Comment (phacks) [inline frontend/src/pages/EnvironmentSettings/Components/AuditParameterTable/AuditParameterTable.style.tsx]: missing a newline here

**Comment by phacks** [comment]
Also, would it possible to align the Close icon in both row types?

image

Thanks!

**Comment by phacks** [comment]
Looks neat, thanks!

It’s not directly related to your PR so we may treat it in a separate issue, but I’ve noticed that all the available configurations are in a selected (or hover?) state even though I’m not interacting with them:

image

Do you know where can it come from?

**Comment by alexfauquette** [comment]
On pages I corrected a bug

When you start writing the URL, after typing the first letter, the focus move back to name input.

To catch modification of the state `isAddin`, I used the solution given in the documentation

https://reactjs.org/docs/hooks-faq.html#how-to-get-the-previous-props-or-state

The bug exists also in "project environement" but it's not a problem since there is only one text input

**Comment by alexfauquette** [comment]
@phacks I resolved the selector problem in PR #48

**Pull request merged**